auth/request/corteza_session_store.go

```diff
@@ -151,23 +154,48 @@ func (s cortezaSessionStore) Save(r *http.Request, w http.ResponseWriter, ses *s
 		ses.ID = string(rand.Bytes(64))
 	}
 
-	if err = s.save(r.Context(), ses); err != nil {
+	var (
+		// make a copy of options for cookie
+		cOpt = *ses.Options
+
+		// expiration for stored session
+		exp time.Time
+	)
+
+	if IsPermLogin(ses) {
+		// if permanent login, make sure max-age is recalculated
+		// so that it slides in the future with every new session-save
+		cOpt.MaxAge = int(s.opt.SessionPermLifetime / time.Second)
+
+		// recalculate expiration for the stored (db) session
+		exp = now().Add(s.opt.SessionPermLifetime)
+	} else {
+		// Warning!
+		// when session is not permanent, max-age is NOT set on the cookie
+		// making cookie last only for the length of the session
+
+		// recalculate expiration for the stored (db) session
+		// we do that even if the cookie does not have a value
+		exp = now().Add(s.opt.SessionLifetime)
```

Comment:
IIRC, we said we'll fallback to `SessionPermLifetime` in case `SessionLifetime` is not set